static/app/views/performance/newTraceDetails/traceApi/useTraceMeta.tsx

```diff
@@ -51,16 +56,37 @@ export function useTraceMeta(
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const mode = queryParams.demo ? 'demo' : undefined;
   const trace = traceSlug ?? params.traceSlug;
 
-  return useApiQuery(
+  const traceMetaQueryResults = useApiQuery<TraceMeta>(
     [
       `/organizations/${organization.slug}/events-trace-meta/${trace ?? ''}/`,
       {query: queryParams},
     ],
     {
       staleTime: Infinity,
-      enabled: !!trace && !!organization.slug,
+      enabled: !!trace && !!organization.slug && mode !== 'demo',
     }
   );
+
+  // When projects don't have performance set up, we allow them to view a sample transaction.
+  // The backend creates the sample transaction, however the trace is created async, so when the
+  // page loads, we cannot guarantee that querying the trace will succeed as it may not have been stored yet.
+  // When this happens, we assemble a fake trace response to only include the transaction that had already been
+  // created and stored already so that the users can visualize in the context of a trace.
+  // The trace meta query has to reflect this by returning a single transaction and project.
+  if (mode === 'demo') {
+    return {
+      data: {
+        errors: 0,
+        performance_issues: 0,
+        projects: 1,
+        transactions: 1,
+      },
+      isLoading: false,
```

Comment:
can we make this conform to the useApiQuery type? At worst, you can return {...traceMetaQueryResult,....} and override the status types, but I would opt not to narrow down the return type as there's good chances we'll need it in the future